src/main/java/net/minecraftforge/fluids/IFluidBlock.java

```diff
@@ -36,12 +35,25 @@
     Fluid getFluid();
 
     /**
+     * Attempts to place the block at a given position. The placed block's level will correspond
+     * to the provided fluid amount.
+     * This method should be called by fluid containers such as buckets, but it is recommended
+     * to use {@link FluidUtil}.
+     *
+     * @param world      the world to place the block in
+     * @param pos        the position to place the block at
+     * @param fluidStack the fluid stack to get the required data from
+     * @param doPlace    if false, the placement will only be simulated
+     * @return the amount of fluid extracted from the provided stack to achieve some fluid level
+     */
+    int place(World world, BlockPos pos, FluidStack fluidStack, boolean doPlace);
```

Comment:
Forge can't break compatibility until the next major version, which can take a while (as far as I know there is no schedule for the next version). If you want this fix in sooner, you will have to do it the less nice way first.